### ISSUE TITLE
Python Rule useless-literal-dict gives false positives when dealing with special characters

### ISSUE THREAD

**Issue by malexmave** [returntocorp/sgrep-rules#1398]
**Describe the bug**
The rule "useless-literal-dicts" seems to strip umlauts from the detected dictionary keys: If two dictionary keys only differ by an Umlaut, they will be seen as equivalent.

**To Reproduce**
1. Go to [the playground for this rule](https://semgrep.dev/editor?registry=python.lang.maintainability.useless-literal.useless-literal-dict)
2. Enter the following source code and evaluate the results:

```python
# Seems to be coerced to {'': '1', '', '2'}
umlauts_dict1 = {'ä': '1', 'ö': '2'}
# Seems to be coerced to {'a': '1', 'a': '2'}
umlauts_dict2 = {'äa': '1', 'öa': '2'}
# Same for accents
umlauts_dict3 = {'áa': '1', 'àa': '2'}
# Even if we explicitly make it a unicode string, same behavior
umlauts_dict4 = {u"áa": '1', u"àa": '2'}
```

**Expected behavior**
None of the above examples should trigger the warning, as the keys are always different.

I am not sure if this is an issue with the rule or with semgrep itself - I could not find any information about semgrep being limited to ASCII with its rules, so I don't think this is expected behavior (if it is, please point me towards the relevant part of the documentation).

**Priority**
How important is this to you?
- [ ] P0: blocking me from making progress
- [ ] P1: this will block me in the near future
- [x] P2: annoying but not blocking me (I can just disable the rule for my CI, I guess?)

**Comment by IagoAbal** [comment]
This is a bug in Semgrep and not a bug in Semgrep-rules.

**Comment by malexmave** [comment]
Thanks - I added the info to a seemingly-relevant issue in the semgrep repo, so we can probably close it here and continue over there.

**Comment by IagoAbal** [comment]
Thanks a lot!
